src/components/PopupWithForm.js

```diff
@@ -1,37 +1,46 @@
-import Popup from './Popup.js';
-
+import Popup from "./Popup.js";
 class PopupWithForm extends Popup {
-    constructor(popupSelector, handleFormSubmit) {
-        super({ popupSelector }); // { popupSelector: "#edit-modal"}
-        this._form = this._popupElement.querySelector('.modal__form');
-        this._popupForm = this._popupElement.querySelector('.modal__form');
-        this._handleFormSubmit = handleFormSubmit;
-        }
+  constructor(popupSelector, handleFormSubmit) {
+    super({ popupSelector });
+    this._form = this._popupElement.querySelector(".modal__form");
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupElement.querySelector(".modal__button");
+    this._handleFormSubmit = handleFormSubmit;
+  }
+
+  _getInputValues() {
+    this._inputList = this._popupElement.querySelectorAll(".modal__input");
+    this._formValues = {};
+    this._inputList.forEach(
+      (input) => (this._formValues[input.name] = input.value)
+    );
+    return this._formValues;
+  }
 
-        _getInputValues() {
-            this._inputList = this._popupElement.querySelectorAll(".modal__input");
-            this._formValues = {};
-            this._inputList.forEach(
-                (input) => (this._formValues[input.name] = input.value)
-            );
-            return this._formValues
-        }
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupElement.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this._handleFormSubmit(this._getInputValues());
+    });
+  }
 
-        setEventListeners() {
-            this._popupElement.addEventListener("submit", (evt) => {
-                this._handleFormSubmit(this._getInputValues());
-                evt.preventDefault();
-            });
-        }
+  open() {
+    super.open();
+  }
 
-    open(){
-        super.open();
-    }    
+  close() {
+    this._popupForm.reset();
+    super.close();
+  }
 
-    close() {
-        this._popupForm.reset()
-        super.close();        
+  renderLoading(loading, loadingText) {
+    console.log("hey is this loading");
+    if (loading) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = "Submit";
```

Comment:
**Could be improved**

It’s better to make a `renderLoading` method inside `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```